modules/data.atmosphere/R/met.process.R

```diff
@@ -47,16 +47,76 @@ met.process <- function(site, input_met, start_date, end_date, model, host, dbpa
 
 
   #read in registration xml for met specific information
-  register.xml <- system.file(paste0("registration/register.", met, ".xml"), package = "PEcAn.data.atmosphere")
-  register <- read.register(register.xml, con)
-
+  met.reg.xml <- system.file(paste0("registration/register.", met, ".xml"), package = "PEcAn.data.atmosphere")
+  met.reg <- read.register(met.reg.xml, con)
+  
+ ##read in registration xml for model specific information
+ ## Where should these registration files be located? Within the model package? This could be difficult given that the package name might not be the same as model name in this context. Will need to double check. 
+  model.reg.xml <- system.file(paste0("registration/register.",model,".xml"), package = paste0("PEcAn.",model))
+  model.reg <- read.register(model.reg.xml, con) #### Betsy: rewrite read.register to accomodate met or model
+    
+  
+ ## Query Database for Input(Site,Source,Machine)
+ input.file.info = db.query(paste("SELECT * from inputs as i join dbfiles as d on i.id = d.container_id where i.site_id = ", 
+                                  site$id,
+                                  "and d.machine_id = ",
+                                  machine$id,
+                                  "and d.container_type = ",
+                                  "'Input'"), con)
+     ##Clean up info Table
+     # Raw  files 
+     raw.info = input.file.info[input.file.info$format_id == met.reg$format$id,]
+     ### grab latest created input record start and end dates
+     raw.clean.info = raw.info[raw.info$updated_at == max(raw.info$updated_at),]
+     ### grab start and end date of record
+     raw.start_date = raw.clean.info$start_date
+     raw.end_date = raw.clean.info$end_date
+     ### grab id of raw record 
+     raw.input.id = raw.clean.info$id
+     
+     # CF files that have raw.id as parent.id
+     cf.info = input.file.info[input.file.info$parent_id %in% raw.input.id,]
+     # Grab file that was last updated
+     cf.clean.info = cf.info[cf.info$updated_at == max(cf.info$updated_at),]
+     ### grab start and end_dates
+     cf.start_date = cf.clean.info$start_date
+     cf.end_date = cf.clean.info$end_date
+     ## grab id of cf record 
+     cf.input.id = cf.clean.info$id
+     
+     # Gapfill files that have cf.id as parent.id
```

Comment:
Regional products don't do gapfilling. This will break the parent.id matching between stages. Also, you're implicitly assuming what occurred during a stage (and thus what needs to occur in the workflow) based solely on depth in the tree, which is going to be very fragile -- not only will this break for regional and regional THREDDS products, but it will also break as soon as Ankur and James add downscaling (and thus additional levels in the processing)